x-pack/plugins/maps/public/store/map.js

```diff
@@ -221,8 +223,13 @@ export function map(state = INITIAL_STATE, action) {
         }
       };
     case SET_SELECTED_LAYER:
-      const match = state.layerList.find(layer => layer.id === action.selectedLayerId);
-      return { ...state, selectedLayerId: match ? action.selectedLayerId : null };
+      const selectedMatch = state.layerList.find(layer => layer.id === action.selectedLayerId);
+      return { ...state, selectedLayerId: selectedMatch ? action.selectedLayerId : null };
+    case ADD_TRANSIENT_LAYER:
```

Comment:
we should call this `SET_TRANSIENT_LAYER` since it's the same concept as `SET_SELECTED_LAYER`